File: src/main/scala/shukaro/artifice/ArtificeMain.scala
package shukaro.artifice

import net.minecraftforge.fml.common.Mod.EventHandler
import net.minecraftforge.fml.common.event._
import net.minecraftforge.fml.common.{Mod, SidedProxy}
import org.apache.logging.log4j.core.Logger
import shukaro.artifice.sided.CommonProxy


@Mod(modid = "Artifice", name = "Artifice", version = "2.0A", modLanguage = "Scala")
object ArtificeMain {

  @SidedProxy(clientSide = "shukaro.artifice.sided.ClientProxy", serverSide = "shukaro.artifice.sided.ServerProxy")
  var proxy: CommonProxy = _

  val logger = new Logger()

  @Mod.Instance
  var instance: ArtificeMain.type = _

  @EventHandler
  def serverStarting(event:FMLServerStartingEvent): Unit = {

  }

  @EventHandler
  def preInit(event: FMLPreInitializationEvent): Unit = {

  }

  @EventHandler
  def init(event: FMLInitializationEvent): Unit = {

  }

  @EventHandler
  def postInit(event: FMLPostInitializationEvent): Unit = {

  }

  @EventHandler
  def imcRequest(event: FMLInterModComms): Unit = {

  }

}

File: src/main/kotlin/shukaro/artifice/ArtificeMain.kt
package shukaro.artifice

import net.minecraftforge.common.config.Configuration
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.common.SidedProxy
import net.minecraftforge.fml.common.event.*
import org.apache.logging.log4j.Logger
import shukaro.artifice.gui.ArtificeTab
import shukaro.artifice.sided.CommonProxy


@Mod(modid = ArtificeMain.MODID, modLanguageAdapter = "net.shadowfacts.forgelin.KotlinAdapter", useMetadata = true)
object ArtificeMain {

    const val MODID = "artifice"

    lateinit var config: Configuration
    lateinit var logger: Logger

    @SidedProxy(clientSide = "shukaro.artifice.sided.ClientProxy", serverSide = "shukaro.artifice.sided.ServerProxy")
    lateinit var proxy: CommonProxy

    val artificeTab: ArtificeTab = ArtificeTab("artifice")

    @Mod.EventHandler
    fun serverStarting(event: FMLServerStartingEvent) {}

    @Mod.EventHandler
    fun preInit(event: FMLPreInitializationEvent) {
        logger = event.modLog
    }

    @Mod.EventHandler
    fun init(event: FMLInitializationEvent) {}

    @Mod.EventHandler
    fun postInit(event: FMLPostInitializationEvent) {}

    @Mod.EventHandler
    fun imcRequest(event: FMLInterModComms) {}

}